selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java

```diff
@@ -152,6 +267,365 @@ public NewWorkspace(
+    return getSamples()
+        .stream()
+        .collect(
+            toMap(element -> getSampleName(element), element -> getSampleDescription(element)));
+  }
+
+  public Set<String> getSamplesNames() {
+    return getSamplesNamesAndDescriptions().keySet();
+  }
+
+  public String getSampleDescription(String sampleName) {
+    return getSamplesNamesAndDescriptions().get(sampleName);
+  }
+
+  public void waitSamplesWithDescriptions(Map<String, String> expectedSamplesWithDescriptions) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>)
+                driver -> expectedSamplesWithDescriptions.equals(getSamplesNamesAndDescriptions()));
+  }
+
+  public void clickOnSampleCheckbox(String sampleName) {
+    String checkboxId = format(CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE, sampleName);
+
+    seleniumWebDriverHelper.waitAndClick(
+        By.xpath(format("//*[@id='%s']/md-checkbox/div", checkboxId)));
+  }
+
+  private void waitSampleCheckboxState(String sampleName, boolean state) {
+    String checkboxId = format(CHECKBOX_BY_SAMPLE_NAME_ID_TEMPLATE, sampleName);
+    String locator = format("//div[@id='%s']/md-checkbox", checkboxId);
+
+    seleniumWebDriverHelper.waitAttributeEqualsTo(
+        By.xpath(locator), "aria-checked", Boolean.toString(state));
+  }
+
+  public void waitSampleCheckboxEnabled(String sampleName) {
+    waitSampleCheckboxState(sampleName, true);
+  }
+
+  public void waitSampleCheckboxDisabled(String sampleName) {
+    waitSampleCheckboxState(sampleName, false);
+  }
+
+  public void waitProjectTabAppearance(String tabName) {
+    seleniumWebDriverHelper.waitVisibility(By.id(tabName));
+  }
+
+  public void waitProjectTabDisappearance(String tabName) {
+    seleniumWebDriverHelper.waitInvisibility(By.id(tabName));
+  }
+
+  public void clickOnProjectTab(String tabName) {
+    waitProjectTabAppearance(tabName);
+    String locator = format(PROJECT_TAB_XPATH_TEMPLATE, tabName);
+    seleniumWebDriverHelper.moveCursorTo(By.xpath(locator));
+
+    waitTextInTooltip(tabName);
+
+    seleniumWebDriverHelper.waitAndClick(By.xpath(locator));
+  }
+
+  private void waitTextInTooltip(String expectedText) {
+    seleniumWebDriverHelper.waitTextEqualsTo(
+        By.xpath("//*[contains(@class, 'tooltip')]"), expectedText);
+  }
+
+  public void waitProjectOptionsForm() {
+    waitProjectNameFieldInProjectOptionsForm();
+    waitDescriptionFieldInProjectOptionsForm();
+    waitRepositoryUrlFieldInProjectOptionsForm();
+  }
+
+  private WebElement waitElementByNameAttribute(String nameAttribute) {
+    return seleniumWebDriverHelper.waitVisibility(
+        By.xpath(format("//input[@name='%s']", nameAttribute)));
+  }
+
+  public WebElement waitProjectNameFieldInProjectOptionsForm() {
+    return waitElementByNameAttribute("projectName");
+  }
+
+  public void waitProjectNameFieldValueInProjectOptionsForm(String expectedValue) {
```

Comment:
If we extract code related to _Project Options Form_ into separate page object, we will be able to get rid of `InProjectOptionsForm` prefix name and it will simplify the code significantly.